src/supermarket/app/services/run_quality_metrics.rb

```diff
@@ -0,0 +1,18 @@
+class RunQualityMetrics
+  attr_reader :cookbook
+
+  def initialize(cookbook)
+    @cookbook = cookbook
+  end
+
+  def self.all_the_latest
+    Cookbook.all.each do |cookbook|
+      new(cookbook).call
```

Comment:
Because we really only need the `id` of the latest version of a cookbook, this almost gets us there:

```
CookbookVersion.where('
  cookbook_versions.created_at = (
    SELECT MAX(latest.created_at) 
    FROM cookbook_versions AS latest 
    WHERE latest.cookbook_id = cookbook_versions.cookbook_id
  )
').pluck(:id)
```

It optimizes around the assumption that the latest cookbook version is the most recently created/published/shared/uploaded. This is not _strictly_ true—our `cookbook.latest_cookbook_version` is implemented as a sort on SemVer—but date-based-latest is probably true enough for a ReRunMetricsOnLatest so that we can make one beefy call to the database and receive the list of CookbookVersion ids.

There's no getting around the need to schedule ~3,200 fieri notifications, though. 😟

![gif-keyboard-10193540256711913767](https://cloud.githubusercontent.com/assets/517302/22596366/44cde280-e9f9-11e6-80f9-a91be84798a3.gif)